lxd/api_cluster.go

```diff
@@ -4585,3 +4602,442 @@ func autoHealCluster(ctx context.Context, s *state.State, offlineMembers []db.No
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	// Update DB entry
+	err = s.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
+		err := dbCluster.DeleteClusterLink(ctx, tx.Tx(), name)
+		if err != nil {
+			return err
+		}
+
+		return err
+	})
+	if err != nil {
+		return response.SmartError(fmt.Errorf("Error deleting %q from database: %w", name, err))
+	}
+
+	requestor := request.CreateRequestor(r)
+	lc := lifecycle.ClusterLinkDeleted.Event(name, requestor, nil)
+	s.Events.SendLifecycle(name, lc)
+
+	return response.SyncResponseLocation(true, nil, lc.Source)
+}
+
+// swagger:operation POST /1.0/cluster/links/{name} cluster-links cluster_link_post
+//
+//	Post cluster link
+//
+//	Link a cluster.
+//
+//	---
+//	produces:
+//	  - application/json
+//	responses:
+//	  "200":
+//	    description: Link cluster request
+//	    schema:
+//	      type: object
+//	      description: Sync response
+//	      properties:
+//	        type:
+//	          type: string
+//	          description: Response type
+//	          example: sync
+//	        status:
+//	          type: string
+//	          description: Status description
+//	          example: Success
+//	        status_code:
+//	          type: integer
+//	          description: Status code
+//	          example: 200
+//	        metadata:
+//	          $ref: "#/definitions/ClusterLinkPost"
+//	  "403":
+//	    $ref: "#/responses/Forbidden"
+//	  "500":
+//	    $ref: "#/responses/InternalServerError"
+func clusterLinkPost(d *Daemon, r *http.Request) response.Response {
+	s := d.State()
+
+	// Parse the request
+	req := api.ClusterLinkPost{}
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return response.BadRequest(err)
+	}
+
+	name, err := url.PathUnescape(mux.Vars(r)["name"])
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	// Request to create a pending link (when a trust token is not provided)
+	if req.TrustToken == "" {
+		networkCert := s.Endpoints.NetworkCert()
+		serverCert := s.ServerCert()
+		notify := newIdentityNotificationFunc(s, r, networkCert, serverCert)
+
+		// Create a pending identity for the cluster link using the identity API
+		identityReq := api.IdentitiesTLSPost{
+			Name:   name,
+			Token:  true,
+			Groups: req.Groups,
+			Type:   api.IdentityTypeCertificateClusterLink,
+		}
+
+		err = s.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
+			// Update DB entry for pending cluster link
+			clusterLink := dbCluster.ClusterLink{
+				Name:        name,
+				Description: req.Description,
+			}
+
+			_, err = dbCluster.CreateClusterLink(ctx, tx.Tx(), clusterLink)
+			if err != nil {
+				return fmt.Errorf("Error inserting %q into database: %w", name, err)
+			}
+
+			return nil
+		})
+		if err != nil {
+			return response.SmartError(err)
+		}
+
+		return createIdentityTLSPending(r.Context(), s, identityReq, notify)
```

Comment:
Refactor this API handler to use the parts of it that you need. Then we can remove the `Type` field from the request body. You need the `Type` field for an internal function call here, but it's not needed for the actual POST /1.0/auth/identities/tls endpoint